### DISCUSSION TITLE
UDM pro for MEO IPTV image freezes

### DISCUSSION THREAD

**Discussion by paulpocarneiro** [fabianishere/udm-iptv#124] [Q&A]
Hello, I would like to replace my MEO FGW with the UDM PRO.
Ideally, I would completely replace it but not having yet the fiber adapter, but for now I would be very happy have IP TV working with the UDM PRO.
I already have FGW on bridge and the UDM connected on port4. so, I have Internet no problem.
I connected a MEO BOX to the UDM and followed the instructions from the link https://github.com/fabianishere/udm-iptv the image works but only for 15 seconds. it then freezes. I change tv chanel and 15 seconds later freezes again.
could anyone please be so kind and give some help?
Many thanks.
paulocarneiro

**Comment by ghost** [comment]
@paulpocarneiro did you found a solution ??

**Reply by jpatriarca** [reply]
You need to active IGMP Snooping for the network where your MEO BOX will be connected. If this is not activated, when the stream moves from unicast (first 10 seconds) to multicast (after that initial 10 seconds) the video stream freezes.

**Reply by ghost** [reply]
Thank you for the answer. And it's turned on. But it still freezes...

**Reply by jpatriarca** [reply]
Where is the MEO BOX connected? Directly to UDM-Pro or connected to a switch that is connected to UDM-Pro? If this switch exists, it is Unifi or other brand?

**Reply by ghost** [reply]
The MEO Box is conected to a US-8-150W. I've tried to conect directly to the UDMP but the result is the same.

**Reply by jpatriarca** [reply]
So, if you have IGMP Snooping activated in the network that will receive the multicast traffic (where you MEO Box is going to be connected) and if even you have connected MEO Box directly connected to UDM-Pro, you have not followed all configurations for installing udm-iptv and respect the suggested firewall rules configuration in your UDM-Pro

**Reply by ghost** [reply]
Hello,

I've done everything but i really cant make it done !!!

i installed like this:

`sh -c "$(curl https://raw.githubusercontent.com/fabianishere/udm-iptv/master/install.sh -sSf)"`

with the conf:

`## Example configuration for udmp-iptv ##`
`# Interface on which IPTV traffic enters the router`
`IPTV_WAN_INTERFACE="eth8.12"`
`# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)`
`IPTV_WAN_VLAN="0"`
`# Name of the IPTV VLAN interface`
`IPTV_WAN_VLAN_INTERFACE="iptv"`
`# IP ranges from which the IPTV traffic originates (separated by spaces)`
`IPTV_WAN_RANGES="10.159.0.0/16  10.173.0.0/16  194.65.46.0/23  213.13.16.0/20  224.0.0.0/4"`
`# DHCP options to send when requesting an IP address`
`IPTV_WAN_DHCP_OPTIONS="-O staticroutes -V IPTV_RG"`
`# LAN interfaces on which IPTV should be made available`
`IPTV_LAN_INTERFACES="br44"`
`# Disable quickleave for igmpproxy`
`IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"`
`# Enable debugging for igmpproxy`
`IPTV_IGMPPROXY_DEBUG="true"`
`IPTV_WAN_DHCP="false"`
`IPTV_IGMPPROXY_PROGRAM="igmpproxy"`
`IPTV_IGMPPROXY_IGMP_VERSION=""`

and configure it to the vlan i've created for IPTV only and then created the firewall rules like this:

`(WAN IN)
- Create New Rule
Type: Internet In
Description: “Allow IGMP Group to LAN
Rule Applied: Before predefined rules
Action: Accept
IPv4 Protocol: All
Destination
IPv4 Address Group: Select the “MEO” address group
Advanced (MANUAL)
check the boxes next to “Match State New”, “Match State Established”, “Match State Invalid” and
“Match State Related”`

`(WAN LOCAL)
- Create New Rule
Type: Internet Local
Description: “Allow IGMP Group to LAN
Rule Applied: Before predefined rules
Action: Accept
IPv4 Protocol: All
Destination
IPv4 Address Group: Select the “MEO” address group
Advanced (MANUAL)
check the boxes next to “Match State New”, “Match State Established”, “Match State Invalid” and
“Match State Related”`

Am i missing something ?

**Reply by ghost** [reply]
`root@ubnt:/# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8.12
WAN VLAN: 0 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 10.159.0.0/16  10.173.0.0/16  194.65.46.0/23  213.13.16.0/20  224.0.0.0/4
LAN Interfaces: br44
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
27: eth8.12@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 2.**.***.***/24 scope global dynamic eth8.12
       valid_lft 3568sec preferred_lft 3568sec
2.XX.XXX.0/24 proto kernel scope link src 2.**.***.***
=== Service Logs ===
Feb 13 19:12:46 ubnt udm-iptvd[2938]: The IGMP message was local multicast. Ignoring.
Feb 13 19:12:46 ubnt udm-iptvd[2938]: RECV V2 member report   from 192.168.44.1    to 224.0.0.22
Feb 13 19:12:46 ubnt udm-iptvd[2938]: The IGMP message was from myself. Ignoring.
Feb 13 19:12:48 ubnt udm-iptvd[2938]: RECV V2 member report   from 192.168.44.6    to 232.21.11.162
Feb 13 19:12:48 ubnt udm-iptvd[2938]: Updated route entry for 232.21.11.162 on VIF #1
Feb 13 19:12:48 ubnt udm-iptvd[2938]: The IGMP message was local multicast. Ignoring.
Feb 13 19:12:49 ubnt udm-iptvd[2938]: RECV V2 member report   from 192.168.44.1    to 224.0.0.2
Feb 13 19:12:49 ubnt udm-iptvd[2938]: The IGMP message was from myself. Ignoring.
Feb 13 19:12:51 ubnt udm-iptvd[2938]: RECV V2 member report   from 192.168.44.6    to 232.0.4.4
Feb 13 19:12:51 ubnt udm-iptvd[2938]: Updated route entry for 232.0.4.4 on VIF #1
`

**Reply by jpatriarca** [reply]
Please explain how do you connect the meo internet to UDM-Pro?
- Do you have a meo router or do you have a ONT?
- Do you have internet with actual configuration (before udm-iptv config)?
- If you have internet, is WAN port configured with VLAN ID?

Check the example below. It is my configuration and I don't need VLAN ID since I have Meo router connected to WAN port.
![image](https://user-images.githubusercontent.com/19894552/218559936-c254eb92-dee8-48a7-a8d8-ab32f3a2bbd0.png)
I suspect you have the configuration like mine. If so, you just need to set the WAN port, where iptv traffic will come, and define which is the bridge your meo box will connect.

**Reply by ghost** [reply]
I have the exactly architecture. I had the meo box running without problems with the USG but since I change to UDM I can't do it. I have this network for years with internet but I don't have the meo box. I'm trying to make it work in this infrastructure but it's not easy. 


        Fiber
          |
    +----------+
    | FTTH NTU |
    +----------+
          |
      VLAN12- IPTV + Internet + VoIP
          |
      +--------+
      | Router |  - UDM PRO
      +--------+
          |
         LAN
          |
      +--------+
      | Switch |  - US-8-150W
      +--------+
       |  |  |
       |  |  +-----------------------------+
       |  |                                |
       |  +-----------------+              |
       |                    |              |
+--------------+       +---------+      +-----+
| IPTV Decoder |       | WiFi AP |      | ... |
+--------------+       +---------+      +-----+
  - MEO IPTV

**Reply by jpatriarca** [reply]
`IPTV_WAN_VLAN="0"`

If I'm not mistaken, if you WAN port have the VLAN tag, you need also to have this setting as VLAN ID 12. Because multicast traffic will enter WAN port tagged, and igmpproxy needs to know which is the VLAN this type of traffic will enter in the network. But you might have already test that possibility, right?

**Reply by ghost** [reply]
> `IPTV_WAN_VLAN="0"`
> 
> If I'm not mistaken, if you WAN port have the VLAN tag, you need also to have this setting as VLAN ID 12. Because multicast traffic will enter WAN port tagged, and igmpproxy needs to know which is the VLAN this type of traffic will enter in the network. But you might have already test that possibility, right?

Yes, i have tried that, but still i cant put it done. I just cant understand why ?!

**Comment by paulpocarneiro** [comment]
Hello all.
Unifi just released UDM PRO SE firmare with support for iptv. I updated bu it still freezes. I have IGMP Snooping on and I have the MEO box directly connected to the UDM PRO SE, wich is connected to the MEO router on port 4 bridge mode. no luck for me. :(
best regards

**Reply by jpatriarca** [reply]
Unfortunately, you're missing some configurations that avoids multicast to work inside your configuration. Got UDM-Pro working with Meo and MEO Box with IGMPProxy in a multi switch architecture. Had you took a look on the firewall rules you need to apply to make multicast subnets to be passed from WAN to your network/VLAN?

**Reply by paulpocarneiro** [reply]
Hi, ups, your absolutely correct. I did not changed any firewall rules. could indicate what to do?
many thanks,

**Reply by jpatriarca** [reply]
Check this topic here: https://github.com/fabianishere/udm-iptv/discussions/119. You'll find a PDF file in the thread. Just follow the steps and create those rules in your firewall configuration. If you follow it, you should be good to correct the freezing issue

**Reply by paulpocarneiro** [reply]
Hello,
I applied the rules and it now works. 
But I was thinking, if I understand correctly these FW rules allow the trafic to all lan, and I would like to only allow to the specific IP of the meo boxes. I created a goup MEO BOXES and how can I adapt the rules to define the trafic only to this group? Considering that in the rules the MEO servers are on the destination I would imagine that I would add the group MEO BOXES to the source but it doesnt work as soon as I updated it the image freazes again. many tanks,

**Reply by jpatriarca** [reply]
I need to check better, but the rules is to bring multicast to all network segment (VLAN) and not a specific IP or group of IPs. Basically, multicast enter to the LAN, IGMPProxy captures that multicast and redirect it to the MEObox. What you’re trying to rule is multicast to be directed to the box, but it don’t work that way. 

An alternative, if you want to isolate that traffic, consider creating a specific VLAN for the IPTV, set up Meo boxes port to that VLAN and do not allow other devices to be added to that VLAN

**Comment by paulpocarneiro** [comment]
Helllo, many thanks, dear jpatricarca any chance I could reach you on message via whatsapp? 
I once tried to change firwall rules and as soon as I define VLAN12 for network MEO all my devices lose internet conectivity. something that I still dont understant what I am doing wrong. I must confess I am still learning.
many thanks,

**Reply by jpatriarca** [reply]
If you're using Meo router in bridge mode (port 4 connected to UDM-Pro WAN port), you don't need to declare VLAN12. This is just when you connect your UDM-Pro WAN port directly to Fiber ONT, but most installation have Meo Router instead. Only in this scenario, you need to declare the VLAN in WAN port configuration. Find @jpatriarca in Telegram and ping me if you need.
